### ISSUE TITLE
TTL/Workspace Schedule is too coupled

### ISSUE THREAD

**Issue by ammario** [coder/coder#2229]
So, I have a workspace that it scheduled to shut down in an hour. I want to edit my workspace schedule (which is a separate idea than extending my lease). I change my TTL in the Edit Schedule to one hour, and then boom, my workspace turns off.

I looked at the code a bit and have no idea what's causing this behavior. 

The fact that "Edit Schedule" modifies my active lease is very hard to reason about as a user. Why does it add the TTL to the lease vs. adding the TTL to my current time? Why would I expect it to change the lease at all?

Also.. we use the word TTL around the product to mean both the schedule and the active lease. This has been confusing as I navigate the code, and also as a user. (See https://github.com/coder/coder/issues/2224 issue, which can be interpreted multiple ways). 

What I suggest is we remove the ability for "Schedule" edits to affect the active lease. A lease is created on Workspace Start and then can only be modified directly. We can expose the ability for the user to edit the active lease separately from the schedule, as described [here](https://github.com/coder/coder/issues/2175).

**Comment by ketang** [comment]
In other words, you want to distinguish between extending this workspace incarnation's lifetime versus extending the lifetime for every future incarnation.

**Comment by johnstcn** [comment]
Thoughts:

> Why would I expect it to change the lease at all?
The "change active lease" behaviour was added in #2165, the idea being that users would not necessarily differentiate between the workspace's "active lease" and the workspace's schedule.

> I looked at the code a bit and have no idea what's causing this behavior.
What likely happened was that the active lease of the workspace had previously been extended, and updating the workspace schedule clobbered this previous lease. This is definitely a bug.

Possible solutions:
1) Make extending the workspace's "active lease" optional and opt-in (UI checkbox / CLI flag)
2) When updating workspace TTL, do not update the deadline of the active workspace build if it had previously been extended. This can be detected by comparing the time the provisioner job was completed plus the current workspace TTL to the deadline of the active workspace build.

We should definitely to 2) in any case; clobbering a previously extended workspace's deadline is not desirable behaviour.

**Comment by ammario** [comment]
@johnstcn I think we added #2165 because (at the time) there was no other way to extend the active lease. With the time-based approach, we can safely eliminate the coupling.

**Comment by johnstcn** [comment]
From discussion:
- When editing a workspace's scheduled shutdown time, the deadline will no longer be updated.
- The time preview will be kept, but the CLI "are you sure?" nag is no longer neccessary. Ditto with the UI danger warning when the time goes down low enough.

**Comment by ketang** [comment]
My understanding (or how I think this should work) is that we only look at the schedule when we create a workspace.  After that, we only look at that workspace's countdown.  Is that right?

**Comment by johnstcn** [comment]
> My understanding (or how I think this should work) is that we only look at the schedule when we create a workspace. After that, we only look at that workspace's countdown. Is that right?

Yes, correct. We used to do it that way but in https://github.com/coder/coder/issues/1783 we decided we should to tightly couple the scheduled shutdown time of a running workspace with the schedule of the workspace -- when the workspace schedule is updated, the actively running workspace's shutdown time also gets updated. 

This is essentially reverting part of the work done in that ticket -- other work done in that ticket is still valuable (showing what the actual shutdown time would be based on changing the schedule).
